src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -38,16 +39,105 @@ namespace Microsoft.ML.Transforms
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The name of the operation in the TensorFlow graph to compute training loss (Optional)", ShortName = "LossOp", SortOrder = 5)]
+            public string LossOperation;
+
+            /// <summary>
+            /// The name of the operation in the TensorFlow graph to compute performance metric during training (Optional).
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The name of the operation in the TensorFlow graph to compute performance metric during training (Optional)", ShortName = "MetricOp", SortOrder = 6)]
+            public string MetricOperation;
+
+            /// <summary>
+            /// Number of samples to use for mini-batch training.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Number of samples to use for mini-batch training.", SortOrder = 7)]
+            public int BatchSize = 64;
+
+            /// <summary>
+            /// Number of training iterations.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Number of training iterations.", SortOrder = 8)]
+            public int Epoch = 5;
+
+            /// <summary>
+            /// The name of the operation in the TensorFlow graph which sets optimizer learning rate (Optional).
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The name of the operation in the TensorFlow graph which sets optimizer learning rate (Optional).", SortOrder = 9)]
+            public string LearningRateOperation;
+
+            /// <summary>
+            /// Learning rate to use during optimization.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Learning rate to use during optimization.", SortOrder = 10)]
+            public float LearningRate = 0.01f;
+
+            /// <summary>
+            /// Shuffle training data on each iteration?
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Shuffle data before each iteration.", SortOrder = 11)]
+            public bool Shuffle = true;
+
+            /// <summary>
+            /// Name of the input in TensorFlow graph that specifiy the location for saving/restoring models to/from disk.
+            /// This parameter is set by different kinds of 'Savers' in TensorFlow and users don't have control over this.
+            /// Therefore, its highly unlikely that this parameter is changed from its default value of 'save/Const'.
+            /// Please change it cautiously if you need to.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Name of the input in TensorFlow graph that specifiy the location for saving/restoring models from disk.", SortOrder = 12)]
+            public string SaveLocationOperation = "save/Const";
```

Comment:
>SaveLocationOperation [](start = 26, length = 21)

would this technique of saving models work if we  modify the graph *structure* ?

i.e. to support a different number of classes (than what the pre-trained model has), suppose i  replace the final layer with new FC layer etc. SO we have a new graph *structure*, and we want to persist the modified graph.  

It seems  using this technique we will end up saving the old graph structure. Am i right ?

Reply:
Current implementation does not change the graph so the same graph works here. Graph can be saved as well but the problem is it is saved in different format which can not be loaded using `FromSavedModel` method.

---
In reply to: [221798703](https://github.com/dotnet/machinelearning/pull/1063#discussion_r221798703) [](ancestors = 221798703)